src/main/java/org/testobject/appium/junit/TestObjectAppiumSuiteWatcher.java

```diff
@@ -0,0 +1,108 @@
+package org.testobject.appium.junit;
+
+import jersey.repackaged.com.google.common.base.Optional;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.testobject.appium.SuiteReporter;
+import org.testobject.rest.api.appium.common.TestObjectCapabilities;
+import org.testobject.rest.api.appium.common.data.SuiteReport;
+import org.testobject.rest.api.appium.common.data.Test;
+import org.testobject.rest.api.appium.common.data.TestReport;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static org.testobject.rest.api.appium.common.TestObjectCapabilities.*;
+
+public class TestObjectAppiumSuiteWatcher extends TestWatcher {
+
+    private String apiKey;
+    private Test test;
+    private boolean isTestLocal;
+    private SuiteReporter reporter;
+    private final URL apiEndpoint;
+
+    public TestObjectAppiumSuiteWatcher() {
+        this.apiEndpoint = TESTOBJECT_API_ENDPOINT;
+    }
+
+    public TestObjectAppiumSuiteWatcher(String apiEndpoint) {
+        try {
+            this.apiEndpoint = new URL(apiEndpoint);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    protected void starting(Description description) {
+        this.test = TestParser.from(description);
+    }
+
+    @Override protected void succeeded(Description description) {
```

Comment:
Usually we put @Override on a separate line to the method name. Applies several times in this class